shell/platform/fuchsia/flutter/session_connection.h

```diff
@@ -94,7 +94,7 @@ class SessionConnection final {
   // The maximum number of frames Flutter sent to Scenic that it can have
   // outstanding at any time. This is equivalent to how many times it has
   // called Present2() before receiving an OnFramePresented() event.
-  static constexpr int kMaxFramesInFlight = 2;
+  static constexpr int kMaxFramesInFlight = 3;
```

Comment:
I am not sure about the nature of scheduling flexibility offered by this change. But, in the past, we have found that increasing the pipeline depth has adverse effects when it comes to render frames with inconsistent rasterizer times.

Reply:
It's important to note that this doesn't actually impact the Flutter pipeline as you normally think of it, but rather how Flutter Present()s content to Scenic.

Right now Flutter only has two frames in flight, which means that if we are effectively "double buffered." If we then miss frame A Flutter can't submit frame C, because B is also in flight. By changing this to 3, we can miss frame A and still have frames B and C in flight too, ensuring that one frame miss won't result in another one.

If we were targeting <=16ms Scenic+Flutter frame build times then this wouldn't matter. But by increasing how many frames in flight we can have we can target 32ms frame build times and not suffer a double penalty for missing a frame.

Reply:
> It's important to note that this doesn't actually impact the Flutter pipeline as you normally think of it, but rather how Flutter Present()s content to Scenic.

I realize that. Just wanted to know the reasoning behind this change. Thanks for the clarifications.